main.py

```diff
@@ -0,0 +1,102 @@
+    )
+    parser.add_argument(
+        "--log",
+        choices=['DEBUG', 'INFO', 'WARNING', 'ERROR', 'CRITICAL'],
+        default="INFO",
+        help="Log level (default: INFO), can be DEBUG, INFO, WARNING, ERROR, CRITICAL",
+    )
+    parser.add_argument(
+        "--preview",
+        action="store_true",
+        help="Enable preview mode. In preview mode, the script will not convert the text to speech. Instead, it will print the chapter index, titles, and character counts.",
+    )
+    parser.add_argument(
+        "--language",
+        default="en-US",
+        help="Language for the text-to-speech service (default: en-US). For Azure TTS (--tts=azure), check https://learn.microsoft.com/en-us/azure/ai-services/speech-service/language-support?tabs=tts#text-to-speech for supported languages. For OpenAI TTS (--tts=openai), their API detects the language automatically. But setting this will also help on splitting the text into chunks with different strategies in this tool, especially for Chinese characters. For Chinese books, use zh-CN, zh-TW, or zh-HK.",
+    )
+    parser.add_argument(
+        "--newline_mode",
+        choices=["single", "double"],
+        default="double",
+        help="Choose the mode of detecting new paragraphs: 'single' or 'double'. 'single' means a single newline character, while 'double' means two consecutive newline characters. (default: double, works for most ebooks but will detect less paragraphs for some ebooks)",
+    )
+    parser.add_argument(
+        "--chapter_start",
+        default=1,
+        type=int,
+        help="Chapter start index (default: 1, starting from 1)",
+    )
+    parser.add_argument(
+        "--chapter_end",
+        default=-1,
+        type=int,
+        help="Chapter end index (default: -1, meaning to the last chapter)",
+    )
+    parser.add_argument(
+        "--output_text",
+        action="store_true",
+        help="Enable Output Text. This will export a plain text file for each chapter specified and write the files to the output folder specified.",
+    )
+    parser.add_argument(
+        "--remove_endnotes",
+        action="store_true",
+        help="This will remove endnote numbers from the end or middle of sentences. This is useful for academic books.",
+    )
+
+    parser.add_argument(
+        "--voice_name",
+        help="Various TTS providers has different voice names, look up for your provider settings.",
+    )
+    parser.add_argument(
+        "--break_duration",
+        default="1250",
+        help="Break duration in milliseconds for the different paragraphs or sections (default: 1250). Valid values range from 0 to 5000 milliseconds.",
+    )
+    parser.add_argument(
+        "--output_format",
+        help="Output format for the text-to-speech service. Supported format depends on selected TTS provider",
```

Comment:
It would be better if we just provide supported options or some links here. Like for Azure, check https://learn.microsoft.com/en-us/azure/ai-services/speech-service/rest-text-to-speech?tabs=streaming#audio-outputs. And for OpenAI, check https://platform.openai.com/docs/guides/text-to-speech/supported-output-formats.

Reply:
There too many supported options, especially for Azure, listing them all together in help snippet of arguments will be too much of text for args help

What will you do when there are more TTS providers and they will also provide their own unique supported formats?
aventually it will be several pages at some stage only of supported formats comments? :)

So I would not list them all specifically

Reply:
OK for now. Maybe add a section in README later.